interpreter/interface.go

```diff
@@ -71,36 +67,39 @@ type StorageWriter interface {
 	) (existed bool)
 }
 
-var _ StorageReader = &Interpreter{}
+var _ StorageWriter = &Interpreter{}
 
 type ValueStaticTypeContext interface {
 	common.MemoryGauge
 	StorageReader
 	TypeConverter
-	SubTypeChecker
+	IsRecovered(location common.Location) bool
```

Comment:
It took me a moment to understand what this meant / where it came from. Maybe add a comment and/or make the naming a bit more verbose, e.g. maybe something like "`IsTypeInfoRecovered`", "`IsElaborationRecovered`", etc.

Reply:
done: https://github.com/onflow/cadence/pull/3822/commits/fed32670917adc8c389b2b2465f2c4f7bf764b02